src/main/java/org/folio/services/ftp/SftpUploadService.java

```diff
@@ -0,0 +1,121 @@
+import org.apache.sshd.sftp.spring.integration.ApacheSshdSftpSessionFactory;
+import org.folio.HttpStatus;
+import org.folio.exceptions.FtpException;
+import org.springframework.integration.file.remote.session.Session;
+import org.springframework.integration.file.remote.session.SessionFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.TimeUnit;
+
+public class SftpUploadService {
+  private static final Logger logger = LogManager.getLogger(SftpUploadService.class);
+  private final String server;
+  private final int port;
+  private static final String FILE_SEPARATOR = "/";
+
+  public SftpUploadService(String uri, Integer portFromConfig) throws URISyntaxException {
+    URI u = new URI(uri);
+    this.server = u.getHost();
+    if (Objects.isNull(portFromConfig)) {
+      portFromConfig = 22;
+    }
+    this.port = u.getPort() > 0 ? u.getPort() : portFromConfig;
+  }
+
+  private ApacheSshdSftpSessionFactory getSshdSessionFactory(String username, String password) throws Exception {
+    var ssh = SshClient.setUpDefaultClient();
+    ssh.start();
+    ApacheSshdSftpSessionFactory factory = new ApacheSshdSftpSessionFactory(false);
+    factory.setHost(server);
+    factory.setPort(port);
+    factory.setUsername(username);
+    factory.setPassword(password);
+    factory.setSshClient(ssh);
+    factory.setConnectTimeout(TimeUnit.SECONDS.toMillis(30L));
+    factory.setAuthenticationTimeout(TimeUnit.SECONDS.toMillis(30L));
+    factory.afterPropertiesSet();
+    return factory;
+  }
+
+  public Future<String> upload(Context ctx, String username, String password, String folder, String filename, String content)
+      throws Exception {
+    Promise<String> promise = Promise.promise();
+    String folderPath = StringUtils.isEmpty(folder) ? "" : (folder + FILE_SEPARATOR);
+    String remoteAbsPath = folderPath + filename;
+
+    SessionFactory<SftpClient.DirEntry> sshdFactory;
+    try {
+      sshdFactory = getSshdSessionFactory(username, password);
+    } catch (Exception e) {
+      throw new FtpException(HttpStatus.HTTP_FORBIDDEN.toInt(),String.format("Unable to connect to %s:%d", server, port));
+    }
+    ctx.owner().executeBlocking(blockingFeature -> {
+      try (InputStream inputStream = new ByteArrayInputStream(content.getBytes()); var session = sshdFactory.getSession()) {
+        logger.info("Start uploading file to SFTP path: {}", remoteAbsPath);
+
+        createRemoteDirectoryIfAbsent(session, folder);
+        session.write(inputStream, remoteAbsPath);
+        logger.info("uploaded: {}", remoteAbsPath);
+        blockingFeature.complete("Uploaded successfully");
+      } catch (Exception e) {
+        logger.info("Error uploading the file {}", remoteAbsPath, e);
```

Comment:
This log should be with error level.
Previous log ("uploaded: {}", remoteAbsPath) could be more informative, you can specify like this: ("File was uploaded to SFTP successfully to path: {}, remoteAbsPath);

The last log "Session closed" can be with debug level, because in any case session will closed because it is finally block and can be changed to  "Sftp session closed"

Reply:
fixed